### PR TITLE
Fix bugs of training

### PR DESCRIPTION
**Pull request by XJDKC**
* Fix bugs of buffering Mult.
* Can train ResNet50 with batch size of 64 now.

### CODE REVIEW

**Comment by XJDKC** [comment]
There is another commit to be pushed.

File: src/core/tensor/tensor.cc [hunk 1]
```diff
@@ -161,6 +161,9 @@ Tensor Resize(const Tensor &in, const Shape &shape) {
 
 // return new tensor
 Tensor Tensor::AsType(const DataType type) {
+  CHECK(block() && block()->initialized() == true)
+      << "the data of the tensor needs be initialized before casting to "
```
Comment (nudles) [inline src/core/tensor/tensor.cc:165]: this cannot fix the error. need to update the test case to reverse the two lines.

Comment (XJDKC) [reply src/core/tensor/tensor.cc:165]: Done.

Comment (XJDKC) [reply src/core/tensor/tensor.cc:165]: There seem to be many such cases in unit testing

**Pull request merged**